pkg/runtime-manager/config/config.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import "strings"
+
+type FailurePolicyType string
+
+const (
+	// PolicyFail returns error to caller when got an error cri hook server
+	PolicyFail FailurePolicyType = "Fail"
+	// PolicyIgnore transfer cri request to containerd/dockerd when got an error to cri serer
+	PolicyIgnore FailurePolicyType = "Ignore"
+)
+
+type RuntimeHookType string
+
+const (
+	defaultRuntimeHookConfigPath string = "/etc/runtime/hookserver.d"
+)
+
+const (
+	PreRunPodSandbox            RuntimeHookType = "PreRunPodSandbox"
+	PreStartContainer           RuntimeHookType = "PreStartContainer"
+	PostStartContainer          RuntimeHookType = "PostStartContainer"
+	PreUpdateContainerResources RuntimeHookType = "PreUpdateContainerResources"
+	PostStopContainer           RuntimeHookType = "PostStopContainer"
+	NoneRuntimeHookType         RuntimeHookType = "NoneRuntimeHookType"
+)
+
+type RuntimeHookConfig struct {
+	RemoteEndpoint string            `json:"remote-endpoint,omitempty"`
+	FailurePolicy  FailurePolicyType `json:"failure-policy,omitempty"`
+	RuntimeHooks   []RuntimeHookType `json:"runtime-hooks,omitempty"`
```

Comment:
each RuntimeHookType may have different FailurePolicyType, maybe better as []struct{RuntimeHooks, FailurePolicy}

Reply:
> 
let's make protocol simple here.
RuntimeHookServer can use different FailurePolicyType for different hooks by:
 set FailurePolicy to 'Failure' and  control grpc response code to nil or error on HookServer side.

Reply:
> each RuntimeHookType may have different FailurePolicyType, maybe better as []struct{RuntimeHooks, FailurePolicy}

let's make protocol simple here.
RuntimeHookServer can use different FailurePolicyType for different hooks by:
set FailurePolicy to 'Failure' and control grpc response code to nil or error on HookServer side.